src/SharedUtilities/implicit_functions.jl

```diff
@@ -0,0 +1,125 @@
+### These functions are here for now, but linsolve!, ldiv!, and
+### thomas_algorithm! will soon be moved to ClimaCore.
+
+import LinearAlgebra
+import ClimaCore: Spaces
+
+export linsolve!, ldiv!, thomas_algorithm!
+
+# Function required by OrdinaryDiffEq.jl
+linsolve!(::Type{Val{:init}}, f, u0; kwargs...) = _linsolve!
+_linsolve!(x, A, b, update_matrix = false; kwargs...) =
+    LinearAlgebra.ldiv!(x, A, b)
+
+# Function required by Krylov.jl (x and b can be AbstractVectors)
+# See https://github.com/JuliaSmoothOptimizers/Krylov.jl/issues/605 for a
+# related issue that requires the same workaround.
+function LinearAlgebra.ldiv!(x, A::AbstractTridiagonalW, b)
+    A.temp1 .= b
+    LinearAlgebra.ldiv!(A.temp2, A, A.temp1)
+    x .= A.temp2
+end
+
+function LinearAlgebra.ldiv!(
+    x::Fields.FieldVector,
+    A::AbstractTridiagonalW,
```

Comment:
maybe this could be RichardsTridiagonalW? not sure :)

Reply:
(related to your comment about hardcoded theta variables in `ldiv!` too)
Maybe we could have `ldiv!` dispatch off of concrete TridiagonalW types (so `RichardsTridiagonalW` here). Then we'll implement one `ldiv!` function for each model (which isn't great), but we would be able to pass the property to access from `x` and `b` (so `soil.\vartheta_l` here) to `_ldiv!/ldiv_serial!`, and keep `_ldiv!`  and `ldiv_serial!` general across all `AbstractTridiagonalW` types.

Reply:
This is a potential solution to #221 - @kmdeck are you okay with addressing this in the PR for that issue?

Reply:
Discussed offline - This will be addressed in the PR linked to #221